search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java

```diff
@@ -224,9 +229,30 @@ public List<GroupEntity> findAllGroups(final GroupQuery query) {
   }
 
   @Override
-  public SearchQueryResult<UserEntity> searchUsers(final UserQuery filter) {
+  public SearchQueryResult<UserEntity> searchUsers(final UserQuery userQuery) {
+    if (userQuery.filter().tenantId() != null) {
+      return searchUsersInTenant(userQuery);
+    }
+
     return getSearchExecutor()
-        .search(filter, io.camunda.webapps.schema.entities.usermanagement.UserEntity.class);
+        .search(userQuery, io.camunda.webapps.schema.entities.usermanagement.UserEntity.class);
+  }
+
+  private SearchQueryResult<UserEntity> searchUsersInTenant(final UserQuery userQuery) {
+    final List<TenantMemberEntity> tenantMembers =
+        getSearchExecutor()
+            .findAll(
+                new TenantQuery.Builder()
+                    .filter(f -> f.joinParentId(userQuery.filter().tenantId()).memberType(USER))
+                    .build(),
+                io.camunda.webapps.schema.entities.usermanagement.TenantMemberEntity.class);
+    final var usernames =
+        tenantMembers.stream().map(TenantMemberEntity::id).collect(Collectors.toSet());
+
+    final var overlayQuery = injectUsernamesIntoQuery(usernames, userQuery);
+
+    return getSearchExecutor()
+        .search(overlayQuery, io.camunda.webapps.schema.entities.usermanagement.UserEntity.class);
```

Comment:
```suggestion
    return searchUsers(overlayQuery);
```
This makes it a bit easier to add other special cases, e.g. filtering users by groups or roles.